beacon-chain/sync/validate_beacon_attestation_test.go

```diff
@@ -163,3 +175,80 @@ func TestValidateBeaconAttestation_Syncing(t *testing.T) {
 		t.Error("Beacon attestation passed validation")
 	}
 }
+
+func TestValidateBeaconAttestation_OldAttestation(t *testing.T) {
+	db := dbtest.SetupDB(t)
+	defer dbtest.TeardownDB(t, db)
+	p := p2ptest.NewTestP2P(t)
+	ctx := context.Background()
+
+	rs := &RegularSync{
+		db: db,
+		chain: &mockChain.ChainService{
+			FinalizedCheckPoint: &ethpb.Checkpoint{
+				Epoch: 10,
+			},
+		},
+		initialSync: &mockSync.Sync{IsSyncing: false},
+	}
+
+	blk := &ethpb.BeaconBlock{
+		Slot: 55,
+	}
+	if err := db.SaveBlock(ctx, blk); err != nil {
+		t.Fatal(err)
+	}
+
+	blockRoot, err := ssz.SigningRoot(blk)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	msg := &ethpb.Attestation{
+		Data: &ethpb.AttestationData{
+			BeaconBlockRoot: blockRoot[:],
+			Source: &ethpb.Checkpoint{
+				Epoch: 10,
+			},
+			Target: &ethpb.Checkpoint{
+				Epoch: 2,
+			},
+		},
+	}
+
+	valid, err := rs.validateBeaconAttestation(ctx, msg, p, false /*fromSelf*/)
+	if err != nil {
+		t.Errorf("Beacon attestation failed validation: %v", err)
+	}
+	if valid {
+		t.Error("Beacon attestation passed validation when it should have failed")
+	}
+
+	if p.BroadcastCalled {
+		t.Error("Message was broadcasted")
+	}
+	// source and target epoch same as finalized checkpoint
+	msg = &ethpb.Attestation{
+		Data: &ethpb.AttestationData{
+			BeaconBlockRoot: blockRoot[:],
+			Source: &ethpb.Checkpoint{
+				Epoch: 10,
+			},
+			Target: &ethpb.Checkpoint{
+				Epoch: 10,
+			},
+		},
+	}
+
```

Comment:
maybe another case for greater then finalized

Reply:
greater than finalized would be the normal case, so we let it pass for further processing